### PR TITLE
Add init containers to PSP admission

### PR DESCRIPTION
**Pull request by smarterclayton**
Treat them just like regular containers.

@pweil-

### CODE REVIEW

**Comment by pweil-** [comment]
LGTM

**Comment by smarterclayton** [comment]
Pushed another commit with the rest of the admission controllers (except for initial resources, which was bigger to refactor.

File: plugin/pkg/admission/securitycontext/scdeny/admission.go [hunk 1]
```diff
@@ -74,6 +74,17 @@ func (p *plugin) Admit(a admission.Attributes) (err error) {
 		return apierrors.NewForbidden(a.GetResource().GroupResource(), pod.Name, fmt.Errorf("SecurityContext.FSGroup is forbidden"))
 	}
 
+	for _, v := range pod.Spec.InitContainers {
+		if v.SecurityContext != nil {
```
Comment (pmorie) [inline plugin/pkg/admission/securitycontext/scdeny/admission.go]: why not extract a method here `denyContainerSecurityContext`?

File: plugin/pkg/admission/serviceaccount/admission.go [hunk 2]
```diff
@@ -324,6 +324,16 @@ func (s *serviceAccount) limitSecretReferences(serviceAccount *api.ServiceAccoun
 		}
 	}
 
+	for _, container := range pod.Spec.InitContainers {
+		for _, env := range container.Env {
```
Comment (pmorie) [inline plugin/pkg/admission/serviceaccount/admission.go]: extract method?

Comment (smarterclayton) [inline plugin/pkg/admission/serviceaccount/admission.go]: Message would be different

**Comment by pweil-** [comment]
Second commit looks fine to me.

@mfojtik @legionus FYI for the next rebase.  Need to sweep our admission plugins.

**Comment by smarterclayton** [comment]
Can I get a label?

File: plugin/pkg/admission/serviceaccount/admission.go (hunk 2 above)
Comment (pmorie) [inline plugin/pkg/admission/serviceaccount/admission.go]: ok, agree it's not worth it here.

File: plugin/pkg/admission/securitycontext/scdeny/admission.go (hunk 1 above)
Comment (pmorie) [inline plugin/pkg/admission/securitycontext/scdeny/admission.go]: You could do it here and technically save lines but it's not worth much.

**Pull request merged**